react-multiple-components/my-app/src/RotatingBanner.tsx

```diff
@@ -0,0 +1,52 @@
+import { Text } from './Text';
+import { Buttons } from './Buttons';
+import React, { useState } from 'react';
+
+type Prop = {
+  items: string[];
+};
+
+export function RotatingBanner({ items }: Prop) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  function handlePrev() {
+    if (currentIndex === 0) {
+      setCurrentIndex(items.length - 1);
+    } else {
+      setCurrentIndex(currentIndex - 1);
+    }
+  }
+
+  function handleNext() {
+    if (currentIndex === items.length - 1) {
+      setCurrentIndex(0);
+    } else {
+      setCurrentIndex(currentIndex + 1);
+    }
+  }
+
+  function handleNumClick(event: React.MouseEvent<HTMLButtonElement>) {
+    const eventTarget = event.target as HTMLButtonElement;
+    const num = eventTarget.textContent;
```

Comment:
So this is why you have the event declared as the parameter to `Buttons.onClick`. This is not a good way to do this because the text might change. Maybe you decided to spell out the numbers, or you make them dots with no text, or you use a language with a different type of number. That would break your code.

You want instead to have Buttons pass the button index in its `onClick` call. Then you can get the number directly from the parameter.